### PR TITLE
[Feature] Add the team_admin role

### LINKED ISSUE #1730: [Feature] Add the team_admin role
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/streampark/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

The team_admin is the super admin inside team(manage team members, operate all apps and projects).

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by 1996fanrui**
## What changes were proposed in this pull request

Issue Number: close #1730 

Add the team_admin role, the Team Admin has all permissions inside the team.

## Brief change log

Add the team_admin role.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): no

### CODE REVIEW

**Comment by 1996fanrui** [comment]
Hi @wolfboys , I have added the team_admin in the PR. Please help take a look in your free time, thanks a lot~

BTW, there is a problem need to be improved:

Currently, the permission isn't changed when switch to other team. For example, user_a is a developer of team_1, and he is a team_admin of team_2. When choose team from team_1 to team_2, the front-end should get the correct permissions from backend.

I created the #1844  to improve it.

**Comment by wolfboys** [comment]
Thanks for your contribution, I'll review it later

File: streampark-console/streampark-console-service/src/assembly/script/data/mysql-data.sql [hunk 1]
```diff
@@ -110,6 +110,7 @@ insert into `t_menu` values (100053, 100001, 'types', null, null, 'user:types',
 -- Records of t_role
 -- ----------------------------
 insert into `t_role` values (100001, 'developer', 'developer', now(), now(), null);
+insert into `t_role` values (100002, 'team_admin', 'Team Admin has all permissions inside the team.', now(), now(), null);
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/assembly/script/data/mysql-data.sql]: how about team_admin change to team_leader?

**Review by wolfboys — COMMENTED** [review]
hi fanrui:

The current pr does not reflect how to assign an team_admin roles to a team (team members are set to the specific implementation of the team_admin, will involve the front-end and back-end), a team is only set up one team_admin or can be set up more than one? Do you have any plans for this?

File: streampark-console/streampark-console-service/src/assembly/script/data/mysql-data.sql (hunk 1 above)
Comment (1996fanrui) [reply streampark-console/streampark-console-service/src/assembly/script/data/mysql-data.sql]: Hi @wolfboys , thanks for your review, it's a good idea.

I have a little concern. To facilitate managing jobs, a team may have multiple `team_admin`, including some developers.

`team_leader` represents the leader of the team. There may be some ambiguity here.

**Comment by 1996fanrui** [comment]
> hi fanrui:
> 
> The current pr does not reflect how to assign an team_admin roles to a team (team members are set to the specific implementation of the team_admin, will involve the front-end and back-end), a team is only set up one team_admin or can be set up more than one? Do you have any plans for this?

Hi @wolfboys 

As I understand, if a team just have a `team_admin`, it cannot guarantee high availability😂. In other word, StreamPark also can have some System Admin.

So I prefer a team can have multiple `team_admin`, and a team can haven't `team_admin`. If we use the plan, the front-end and back-end don't need change any code.

**Review by wolfboys — APPROVED** [review]
LGTM
Thanks for your hard work, great feature.  BTW, The use documents of team also need to be supplemented!

**Comment by 1996fanrui** [comment]
> LGTM Thanks for your hard work, great feature. BTW, The use documents of team also need to be supplemented!

@wolfboys Thanks for your review and reminder. I will start sort out the team related doc next week.

**Pull request merged**